packages/web-components/fast-foundation/src/disclosure/disclosure.spec.md

```diff
@@ -0,0 +1,98 @@
+# Disclosure
+
+## Overview
+
+A disclosure component is the combination of a button and section of content, a button that toggles the visibility of the section. As defined by the W3C:
+
+> A disclosure is a button that controls the visibility of a section of content. When the controlled content is hidden, it is often styled as a typical push button with a right-pointing arrow or triangle to hint that activating the button will display additional content. When the content is visible, the arrow or triangle typically points down.
+
+### Use Cases
+
+-   In general, reveal the extra information on button click when the end-user is needed to know/see.
+-   Laurel wanted to buy a new phone and she was looking at one of the websites where all the basic information was shown. She read about a camera feature called Night Mode and there was a section labeled "See the advanced technology that goes into every Night mode shot". She clicked on it and found out how that feature would work. With that extra information on that feature, she was quite confident in her buying.
+
+### Features
+
+-   Enter and Space: activates the disclosure component and toggles the visibility of the disclosure content.
+-   Use `expanded` attribute or `toggle()` to render default expanded
+-   Observe the state when the disclosure is toggled
+-   `show()` and `hide()` as helper methods to toggle the content from outside
+
+### Prior Art/Examples
+
+-   [Shopify Polaris](https://polaris.shopify.com/components/behavior/collapsible)
+-   [Lightning Design](https://www.lightningdesignsystem.com/components/expandable-section/)
+-   [Bootstrap](https://getbootstrap.com/docs/4.3/components/collapse/)
+-   [Material UI](https://material-ui.com/api/collapse/)
+
+---
+
+## Design
+
+The disclosure component can easily be extended for customization, for example adding animation to reveal the content or get the height of extra content slot.
+
+-   `showAnimation({ contentNode })` and `hideAnimation({ contentNode })` `async` methods are `protected` and designed to override in the derived class to add animations.
```

Comment:
I'd like to learn a bit more about the animation thinking. Could this be done with css instead for example?

Reply:
For example, if we gave the content a part name and added additional part names dynamically in different states, could a css animation be targeted at the part to transition between open and close?

Reply:
I tried CSS earlier e.g. set `max-height: 100vh`  and apply the desired transition or animation effect for the extra content toggling, however, `100vh` would consider complete viewport height and extra content size is always going to be varied (could be `100vh` or `10px` too). Therefore, while animating the consideration of `100vh` is better to be avoided and calculate the actual height of the content.

In addition, when the transition is in the process there can be a control to disable the button or wait for the first transition to be finished, etc than queueing.

The `fast-foundation` component would just do toggling of the extra content and the `fast-component` or any custom component that extends the foundation component would take care of the animation.

Because of the height calculation being done dynamically, I am not sure if we could target the CSS animation at the part to transition between open and close.🤔